x-pack/plugins/security_solution/public/cloud_security_posture/components/csp_details/insights_tab_csp.tsx

```diff
@@ -5,19 +5,105 @@
  * 2.0.
  */
 
-import React, { memo } from 'react';
-import { EuiSpacer } from '@elastic/eui';
+import React, { memo, useMemo, useState } from 'react';
+import type { EuiButtonGroupOptionProps } from '@elastic/eui';
+import { EuiButtonGroup, EuiSpacer } from '@elastic/eui';
+import { FormattedMessage } from '@kbn/i18n-react';
+import type { FlyoutPanelProps, PanelPath } from '@kbn/expandable-flyout';
+import { useExpandableFlyoutState } from '@kbn/expandable-flyout';
+import { i18n } from '@kbn/i18n';
+import type { FlyoutPanels } from '@kbn/expandable-flyout/src/store/state';
 import { MisconfigurationFindingsDetailsTable } from './misconfiguration_findings_details_table';
+import { VulnerabilitiesFindingsDetailsTable } from './vulnerabilities_findings_details_table';
 
 /**
  * Insights view displayed in the document details expandable flyout left section
  */
+
+interface CspFlyoutPanelProps extends FlyoutPanelProps {
+  params: {
+    path: PanelPath;
+    hasMisconfigurationFindings: boolean;
+    hasVulnerabilitiesFindings: boolean;
+  };
+}
+interface CspFlyoutPanels extends FlyoutPanels {
+  left: CspFlyoutPanelProps;
+}
+
 export const InsightsTabCsp = memo(
   ({ name, fieldName }: { name: string; fieldName: 'host.name' | 'user.name' }) => {
+    const panels = useExpandableFlyoutState() as CspFlyoutPanels;
```

Comment:
Assertions shouldn't really be used outside of tests or really special cases. if the function does not return a proper `CspFlyoutPanels` it means something can break and we are just hiding it by asserting a type

Reply:
I agree with @JordanSh here, we should avoid using assertions whenever possible, I think we can make better use of type guards here instead of using assertion

```ts
interface CspFlyoutPanelProps extends FlyoutPanelProps {
  params: {
    path?: PanelPath;
    hasMisconfigurationFindings: boolean;
    hasVulnerabilitiesFindings: boolean;
  };
}

// Type guard to check if the panel is a CspFlyoutPanelProps
function isCspFlyoutPanelProps(
  panelLeft: FlyoutPanelProps | undefined
): panelLeft is CspFlyoutPanelProps {
  return (
    !!panelLeft?.params?.hasMisconfigurationFindings ||
    !!panelLeft?.params?.hasVulnerabilitiesFindings
  );
}
```

Reply:
Following my previous comment, we can make use of default initializers to use type guards checking without breaking react hooks rules:


```ts
    const panels = useExpandableFlyoutState();

    let hasMisconfigurationFindings = false;
    let hasVulnerabilitiesFindings = false;
    let subTab: string | undefined;

    // Check if panels.left is of type CspFlyoutPanelProps and extract values
    if (isCspFlyoutPanelProps(panels.left)) {
      hasMisconfigurationFindings = panels.left.params.hasMisconfigurationFindings;
      hasVulnerabilitiesFindings = panels.left.params.hasVulnerabilitiesFindings;
      subTab = panels.left.params.path?.subTab;
    }

    const getDefaultTab = () => {
      if (subTab) {
        return subTab;
      }

      return hasMisconfigurationFindings
        ? CspInsightLeftPanelSubTab.MISCONFIGURATIONS
        : hasVulnerabilitiesFindings
        ? CspInsightLeftPanelSubTab.VULNERABILITIES
        : '';
    };

    const [activeInsightsId, setActiveInsightsId] = useState(getDefaultTab());
```